### PR TITLE
Add optional authentication for qblast

### PR DESCRIPTION
**Pull request by sorsaffari**
[NSBI BLAST on the cloud](https://aws.amazon.com/marketplace/pp/B00N44P7L6?qid=1525720065294&sr=0-1&ref_=srh_res_product_title) by default requires authentication.

This pull request allows the developer to pass the `username` and `password` as parameters of the `qblast()` method. The credentials are then used in creating an [HTTPBasicAuthHandler](https://docs.python.org/3.1/howto/urllib2.html#id6) that in turn creates and authenticated `opener` for requests against a BLAST instance on the cloud.

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file and understand that AppVeyor and
TravisCI will be used to confirm the Biopython unit tests and ``flake8`` style
checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

File: Bio/_py3k/__init__.py [hunk 1]
```diff
@@ -131,7 +131,7 @@ def _binary_to_string_handle(handle):
     from io import StringIO
 
     # On Python 3 urllib, urllib2, and urlparse were merged:
-    from urllib.request import urlopen, Request, urlretrieve, urlparse, urlcleanup
+    from urllib.request import urlopen, Request, urlretrieve, urlparse, urlcleanup, HTTPPasswordMgrWithDefaultRealm, HTTPBasicAuthHandler, build_opener, install_opener
```
Comment (peterjc) [inline Bio/_py3k/__init__.py]: Where do you import these from on Python 2.7?

**Comment by peterjc** [comment]
We've been lucky so far in Python 2 vs 3 differences, but this has likely been been in part from avoiding new things in Python 3 only. I don't know enough about urllib credentials to have an expert opinion here.

**Comment by mdehoon** [comment]
It seems that remote Blast has moved to the cloud, and qblast doesn't exist any more.

This page:

https://ncbi.github.io/blast-cloud/

calls the Blast AMI deprecated and no longer supported, and `python test_NCBI_qblast.py` has not given me a response.

**Comment by MarkusPiotrowski** [comment]
I think their web api is still working. Takes some time (as usual).
grafik

**Comment by mdehoon** [comment]
superseded by #4149

**Pull request closed without merging**